python/tests/integration/test_error_handling.py

```diff
@@ -0,0 +1,151 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+import pytest
+import tornado
+import tornado.gen
+
+from tchannel.exceptions import ProtocolException
+from tchannel.messages.call_request import CallRequestMessage
+from tchannel.messages.call_request_continue import CallRequestContinueMessage
+from tchannel.messages.common import ChecksumType
+from tchannel.messages.common import FlagsType
+from tchannel.tornado import RequestDispatcher
+from tchannel.tornado import TChannel
+from tchannel.tornado.connection import StreamConnection
+from tests.integration.server_manager import TChannelServerManager
+
+
+@tornado.gen.coroutine
+def handler1(request, response, proxy):
+    raise Exception("application uncaught exception")
+
+
+@tornado.gen.coroutine
+def handler2(request, response, proxy):
+    response.set_header_s(request.get_header_s())
+    response.set_body_s(request.get_body_s())
+
+
+@pytest.fixture
+def handlers():
+    dispatcher = RequestDispatcher()
+    dispatcher.register("endpoint1", handler1)
+    dispatcher.register("endpoint2", handler2)
+    return dispatcher
+
+
+@pytest.yield_fixture
+def tchannel_server(random_open_port, handlers):
+    with TChannelServerManager(
+            port=random_open_port,
+            dispatcher=handlers) as manager:
+        yield manager
+
+
+@pytest.mark.gen_test
+def test_unexpected_error_from_handler(tchannel_server):
+    # test for invalid call request message
+    hostport = 'localhost:%d' % tchannel_server.port
+    tchannel = TChannel()
+    connection = yield StreamConnection.outgoing(
+        hostport=hostport,
+        tchannel=tchannel,
+    )
+
+    callrequest = CallRequestMessage(
+        flags=FlagsType.fragment,
+        args=[
+            'endpoint1',
+            '',
+            '',
+        ]
+    )
+    # set a wrong checksum
+    callrequest.checksum = (ChecksumType.crc32c, 1)
+    with pytest.raises(ProtocolException):
+        yield connection.send(callrequest)
+
+
+@pytest.mark.gen_test
+def test_invalid_message_during_streaming1(tchannel_server):
+    # test for invalid call request message
+    hostport = 'localhost:%d' % tchannel_server.port
+    tchannel = TChannel()
+    connection = yield StreamConnection.outgoing(
+        hostport=hostport,
+        tchannel=tchannel,
+    )
+
+    callrequest = CallRequestMessage(
+        flags=FlagsType.fragment,
+        args=[
+            'endpoint2',
+            'a',
+            'a',
+        ],
+        headers={'as': 'raw'},
+    )
+
+    callreqcontinue = CallRequestContinueMessage(
+        flags=FlagsType.fragment,
+        args=[
+            'a',
+        ],
+    )
+
+    with pytest.raises(ProtocolException) as e:
```

Comment:
This is wrapping a pretty large block of code, just put this around the statement you expect to raise the error.